distsql/distsql.go

```diff
@@ -32,16 +32,14 @@ var (
 )
 
 var (
-	_ SelectResult  = &selectResult{}
-	_ PartialResult = &partialResult{}
+	_ NewSelectResult  = &newSelectResult{}
+	_ NewPartialResult = &newPartialResult{}
 )
 
-var selectResultGP = gp.New(2 * time.Minute)
-
-// SelectResult is an iterator of coprocessor partial results.
-type SelectResult interface {
+// NewSelectResult is an iterator of coprocessor partial results.
```

Comment:
Why not just remove the `New` key word?

Reply:
it's too many things to change, do it in next pr....

Reply:
ok